intro-to-R-tidyverse/00b-debugging_resources.md

```diff
@@ -0,0 +1,244 @@
+[`intro_to_R` module](https://alexslemonade.github.io/training-modules/intro-to-R-tidyverse/01-intro_to_r.nb.html),
+objects have *structures* and *types*. Having input that doesn't match the 
+requirements that a function has can be a common source of errors. Pay special
+attention to what the documentation says about what kind of input and output 
+the function is designed to use. 
+
+#### Use the RStudio help bar
+
+Here's a screenshot from the help window in RStudio. Note that here we searched
+for the `levels` function. R documentation includes information about what the
+expected arguments are as well as examples of how to use a function. Note here 
+that this documentation tells us that the input for `x` is probably a factor. 
+![search_bar](screenshots/r_search_bar.png)
+
+#### For Bioconductor package functions, look at their documents
+
+Like other a lot of other packages we may be using for genomics analysis, [DESeq2](https://www.bioconductor.org/packages/release/bioc/html/DESeq2.html) is 
+a Bioconductor package. We use DESeq2 in the bulk RNA-seq module. 
+
+The documentation on Bioconductor pages have information that can be valuable 
+for troubleshooting. Every bioconductor package has a PDF reference where all the 
+functions and objects for that package are described. They can take some getting
+used to, but generally can have helpful information. 
+
+### 6) Google it again
+
+Because it's unlikely your first attempt at Googling will lead you straight
+to an answer; this is something you should continue to try with different wordings. 
+Through trial and error, and also Google algorithms learning about what you look
+for, your search results can eventually lead you to helpful examples and forums.
+
+### 7) Look at the source code for that function
+
+This should rarely be your first approach to solving a problem, since this
+approach is difficult and doesn't always pay off. 
+This approach will require a a bit more practice at reading code, so it
+may not be the most fruitful approach depending on the readability and 
+complexity of the code. 
+
+### 8) Post to an appropriate forum on StackOverflow or a GitHub Issue
+
+After you've tediously mined the internet for solutions to your problem and 
+still not resolved your problem, you can post your problem to the internet for
+help. 
+
+*Keys for asking for coding help are:* 
+- Be *specific* with what the problem is. 
+- Show them the code for what you have already tried.
+- Give a workable example. See this excellent [StackOverflow post](https://stackoverflow.com/questions/5963269/how-to-make-a-great-r-reproducible-example/5963610#5963610) 
+on best practices for doing this for R.
+- Show your set up environment, including your session info. 
+
+For more on how to ask for programming help in general, [see this article](https://www.propublica.org/nerds/how-to-ask-programming-questions).
+The better you are able to follow these advice in posting your question, the 
```

Comment:
Advice is apparently what is called a mass noun ([ref](https://jakubmarian.com/advice-or-advices-singular-or-plural/))
```suggestion
The better you are able to follow this advice in posting your question, the 
```